Test/GitHubRelease.Tests.ps1

```diff
@@ -0,0 +1,573 @@
+
+#Requires -Version 4
+Set-StrictMode -Version 'Latest'
+
+& (Join-Path -Path $PSScriptRoot -ChildPath 'Initialize-WhiskeyTest.ps1' -Resolve)
+
+$apiKeys = @{ }
+$failed = $false
+
+function GivenApiKey
+{
+    param(
+        $Name,
+        $Value
+    )
+
+    $apiKeys[$Name] = $Value
+}
+
+function GivenAssets
+{
+    param(
+        $Asset
+    )
+
+    $assetJson = ConvertTo-Json -InputObject $Asset
+    Mock -CommandName 'Invoke-RestMethod' `
+         -ModuleName 'Whiskey' `
+         -ParameterFilter { $Uri -like '*/releases/*/assets' } `
+         -MockWith ([scriptblock]::Create("'$assetJson' | ConvertFrom-Json"))

```

Comment:
Because with a script block, the `$assetJson` variable wouldn't get evaluated until the script block executes, which is in the `GitHubRelease` task, where it doesn't exist. If I create a string of PowerShell, I capture the value of the `$assetJson` variable as it in this context. And `.GetNewClosure()` on the script block doesn't work.

Reply:
Ah ok I see. Thanks for the explanation.